src/services/products.service.ts

```diff
@@ -1,68 +1,221 @@
+      this.mergeCleanerFeatures(tier, mergedFeatures);
+    }
+  }
 
-        if (tiers.length === 0) {
-          throw new TierNotFoundError(`No tiers found for user uuid ${userUuid} or associated workspaces.`);
+  private mergeMailFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const mailFeatures = tier.featuresPerService[Service.Mail];
+    if (mailFeatures.enabled && mailFeatures.addressesPerUser > mergedFeatures.mail.addressesPerUser) {
+      mergedFeatures.mail = {
+        enabled: true,
+        addressesPerUser: mailFeatures.addressesPerUser,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeMeetFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const meetFeatures = tier.featuresPerService[Service.Meet];
+    if (meetFeatures.enabled && meetFeatures.paxPerCall > mergedFeatures.meet.paxPerCall) {
+      mergedFeatures.meet = {
+        enabled: true,
+        paxPerCall: meetFeatures.paxPerCall,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeVpnFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const vpnFeatures = tier.featuresPerService[Service.Vpn];
+    // We cannot determine which featureId provides access to more zones
+    // maybe add this info to the tier object?
+    if (vpnFeatures.enabled && !mergedFeatures.vpn.enabled) {
+      mergedFeatures.vpn = {
+        enabled: true,
+        featureId: vpnFeatures.featureId,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeAntivirusFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const antivirusFeatures = tier.featuresPerService[Service.Antivirus];
+    if (antivirusFeatures.enabled && !mergedFeatures.antivirus.enabled) {
+      mergedFeatures.antivirus = {
+        enabled: true,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeBackupsFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const backupsFeatures = tier.featuresPerService[Service.Backups];
+    if (backupsFeatures.enabled && !mergedFeatures.backups.enabled) {
+      mergedFeatures.backups = {
+        enabled: true,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeCleanerFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const cleanerFeatures = tier.featuresPerService[Service.Cleaner];
+    if (cleanerFeatures.enabled && !mergedFeatures.cleaner.enabled) {
+      mergedFeatures.cleaner = {
+        enabled: true,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeFeatures(tiers: Tier[]): MergedTierFeatures {
+    const mergedFeatures: MergedTierFeatures = {
+      mail: { enabled: false, addressesPerUser: 0 },
+      meet: { enabled: false, paxPerCall: 0 },
+      vpn: { enabled: false, featureId: '' },
+      antivirus: { enabled: false },
+      backups: { enabled: false },
+      cleaner: { enabled: false },
+      drive: { tier: tiers[0] },
```

Comment:
Why do you return the entire tier for drive? isn't it enough returning `tiers[0].featuresPerService[Service.Drive]`?